### PR TITLE
llvm 10.0.0

### PR DESCRIPTION
**Pull request by smorimoto**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by imbsky** [comment]
The test succeeds, but there is a problem. I don't know what approach to take in brew to avoid this problem, so I would like to hear the opinion.
```
λ brew audit --strict llvm
llvm:
  * python modules have explicit framework links
    These python extension modules were linked directly to a Python
    framework binary. They should be linked with -undefined dynamic_lookup
    instead of -lpython or -framework Python.
      /usr/local/opt/llvm/lib/python2.7/site-packages/lldb/_lldb.so
```

**Comment by SMillerDev** [comment]
Please leave LLVM@9 out for now. You could make a new pull request for it if needed.

**Comment by Bo98** [comment]
Can you reduce the diff on `llvm`? You moved blocks around for no reason and actually in ways that make the audit fail:

```
==> brew audit llvm --online
==> FAILED
Error: 3 problems in 1 formula detected
llvm:
  * C: 19: col 3: `stable` (line 19) should be put before `pour_bottle?` (line 6)
  * C: 69: col 3: `head` (line 69) should be put before `keg_only` (line 11)
```

Also `oclgrind` will need a revision bump.

**Comment by timmyjose** [comment]
Please expedite this! :-)

**Comment by SMillerDev** [comment]
Please consider hosting a copy in a personal tap if you want this done fast, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

If you want it done well you'll have to wait for maintainers to have time to review the changes and for CI to test it.

**Comment by Bo98** [comment]
oclgrind might just need a revision bump rather than `llvm@9`, unless you tried it out locally and it didn't work.

**Comment by timmyjose** [comment]
@SMillerDev It was a friendly suggestion. I very well know that it takes time to review and test - no need to get snarky.

**Review by Bo98 — CHANGES_REQUESTED** [review]
* Remove `Aliases/llvm@9`. You don't need an alias to something of the same name.

```
Error: Formula duplicating alias: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/llvm@9
```

* It looks like `oclgrind` does in fact need `llvm@9`, but still keep the revision bump as well since it will be needed to ensure users have the right dependency.

* The test of `llvm@9` fails. Here's what I think will fix it:

File: Formula/llvm@9.rb
Comment (Bo98) [inline Formula/llvm@9.rb]:
```suggestion
    inreplace "#{share}/clang/tools/scan-build/bin/scan-build", "$RealBin/bin/clang", "#{bin}/clang".gsub("@", "\\@")
```

**Comment by imbsky** [comment]
Thank you for the review. I applied that.

**Comment by bcardiff** [comment]
Thanks @imbsky for this PR!

The current `Formula/crystal.rb` will also need `depends_on "llvm@9"`. In the next version it will have support for llvm 10. Should that be fixed in this pr as `Formula/oclgrid.cr` or in a separate one?

**Comment by Bo98** [comment]
Probably this one. Can the test be improved to pick up those cases better? It seems like the current one never fails if the LLVM version is incompatible.

**Comment by bcardiff** [comment]
I will try to improve them for some smoke testing. Because right now the issue appears when building from source probably or running the compiler specs that are to heavy for the CI.

**Comment by Bo98** [comment]
I could potentially add `llvm` to the list of formulae which test building dependents from source, if that doesn't end up being too costly time-wise.

**Comment by iMichka** [comment]
```
llvm@9:
  * Versioned formulae in homebrew/core should use `keg_only :versioned_formula`
  * Formulae should not have a `HEAD` spec
```

@imbsky can you please run `brew audit llvm@9 --online --new-formula llvm@9` and `brew audit llvm@9 --online llvm` before pushing any changes? Running this build block our CI for 5 to 6 hours each time, so it would be great if audit errors are fixed before pushing. Thanks.

**Comment by imbsky** [comment]
Even in my own environment, it takes a lot of time, so to be honest, it's not very practical to try every time. Since I changed to use Ninja, it's about 20% faster, but it still takes about 1 hour for every build.

**Comment by imbsky** [comment]
I'm sorry, I made some unnecessary commits, so please cancel them manually if necessary. I will try to test it in advance from next time. The only remaining problem is the following. Does anyone have a solution?

```
λ brew audit --online llvm
llvm:
  * python modules have explicit framework links
    These python extension modules were linked directly to a Python
    framework binary. They should be linked with -undefined dynamic_lookup
    instead of -lpython or -framework Python.
      /usr/local/opt/llvm/lib/python2.7/site-packages/lldb/_lldb.so
Error: 1 problem in 1 formula detected
```

**Comment by imbsky** [comment]
The last commit should pass the test.

**Comment by imbsky** [comment]
It seems that the build was aborted for some reason.

**Comment by Bo98** [comment]
It's already requeued. There were some CI issues earlier.

**Comment by imbsky** [comment]
I wasn't used to Jenkins so I didn't know how to check it. Thank you!

**Comment by imbsky** [comment]
All tests have passed!

**Comment by imbsky** [comment]
@SMillerDev @Bo98 @iMichka Could you check this?

**Comment by Bo98** [comment]
I think this is good. I checked the bottle earlier and nothing seems to be missing. I'm going to check `crystal` in a moment and then I'll merge this.

**Comment by imbsky** [comment]
@Bo98 Thank you!

**Comment by imbsky** [comment]
When I built the Crystal compiler locally with LLVM 10.0.0 (brew version), there seemed to be no problems.

**Comment by bcardiff** [comment]
Lucky us! It is usually a problem. If something arises, I will fix it later, we should be near to release the next version with llvm 10 support though.

**Review by Bo98 — APPROVED** [review]

**Comment by Bo98** [comment]
Thanks!

**Comment by imbsky** [comment]
@Bo98 Thank you, too!

**Comment by timmyjose** [comment]
Thank you to @imbsky and all the folks who worked on this - I had built llvm+lld+clang 10 on my own in the interim, but it took around 40GB (!) on my local machine, so this is a very welcome release. Cheers!

**Pull request closed without merging**
